portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,36 +23,70 @@
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.util.*;
+import java.text.*;
+import java.sql.Timestamp;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;  
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private ArrayList<String> messages;
-
-    @Override
-    public void init() {
-        messages= new ArrayList<String>();
-    }
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+Query query = new Query("Task");
+System.out.println("query:"+query);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+System.out.println("d:"+datastore);
+        PreparedQuery results = datastore.prepare(query);
+System.out.println("r:"+results);
+        List<Comment> comments = new ArrayList<>();
         
-        //converting ArrayList to json string 
-        Gson gson = new Gson();
-        String json = gson.toJson(messages);
+        for (Entity entity : results.asIterable()) {
+            Key id = entity.getKey();
+System.out.println("key:"+entity.getKey());            
+            String title = (String) entity.getProperty("title");
+System.out.println("id:"+id+" title:"+title);
+            long timestamp = (long) entity.getProperty("timestamp");
+System.out.println("timestamp:"+timestamp);
+            Date date = new Date(timestamp);
+            Format format = new SimpleDateFormat("dd-MM-yyy HH:mm");
+            String dateAndTime =format.format(date);
+System.out.println("d&t:"+dateAndTime);
 
-        // Send the JSON as the response
+            Comment comment = new Comment(id, title, dateAndTime);
+            comments.add(comment);
+System.out.println("cArray:"+comments);            
+        }
+
+        Gson gson = new Gson();
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(gson.toJson(comments));
+
     }
 
     @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String comment = request.getParameter("comment");
-      messages.add(comment);
-    
-      // Redirect back to the HTML page.
-      response.sendRedirect("/index.html#add-comments");
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String comment = request.getParameter("comment");
```

Comment:
It would be nice if we could check the input before using. For example, we could check if "comment" parameter has been set.

Reply:
Ok, so I'll place an if statement to check that and then send an error if comment has not been set.

Reply:
But I placed 'required' as a condition in my form, so that it isn't submitted if the 'comment' is null.
I thought it would be enough, but again its good to be sure.

Reply:
Verification always has to be done server side. If you do client-side verification, you still have to repeat it on the server. Otherwise, your web app cannot be secure as people can access your API directly skipping any security measures which are implemented on the client side.

Reply:
Oh, thanks for letting me know. I should have known that :E